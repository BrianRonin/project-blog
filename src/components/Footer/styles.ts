import styled, { css } from 'styled-components'
import { Main as Text } from '../Text/styles'

export const Main = styled.div`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    background-color: white;
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    & ${Text} {
      font-size: ${theme.fonts.sizes.small};
    }
  `}
`
