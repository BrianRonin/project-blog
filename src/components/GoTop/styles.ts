import styled, { css, DefaultTheme } from 'styled-components'

const Light = (theme: DefaultTheme) => css`
  color: ${theme.colors.primary};
  background: ${theme.colors.white};
`
type background = {
  background: boolean
}

export const Main = styled.a<background>`
  ${({ theme, background }) => css`
    position: fixed;
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.8;
    border-radius: 50%;
    ${background && Light(theme)};
  `}
`
