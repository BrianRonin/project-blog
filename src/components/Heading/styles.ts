import styled, { css, DefaultTheme } from 'styled-components'
import { HeadingProps } from '.'

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.large};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xlarge};
    @media ${theme.media.medium} {
      font-size: ${theme.sizes.large};
    }
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xhuge};
    ${mediaFont(theme)}
  `,
}

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.sizes.xlarge};
  }
`

export const Title = styled.h1<
  Pick<HeadingProps, 'light' | 'size' | 'uppercase'>
>`
  ${({ theme, light, size, uppercase }) => css`
    color: ${light ? theme.colors.white : theme.colors.primary};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
    ${!!size && titleSize[size](theme)};
  `}
`
