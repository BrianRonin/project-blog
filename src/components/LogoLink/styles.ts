import styled from 'styled-components'
export const Main = styled.a`
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;

  > img {
    width: 7rem;

    border-radius: 50%;
  }
`

export const Logo = styled.img`
  max-height: 6rem;
`
