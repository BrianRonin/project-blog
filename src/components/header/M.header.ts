import { headerProps } from './header'

export const mock_header = {
  title: 'um titulo',
  description: '<p>description lorem ipsum banana couve ma√ßa<p/>',
  alt: 'alt da imagem',
  srcImg: '/assets/images/piramide.png',
} as headerProps
