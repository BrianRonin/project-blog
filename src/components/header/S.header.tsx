import styled, { css, DefaultTheme } from 'styled-components'
import { Title } from '../Heading/styles'

const name = (theme: DefaultTheme) => css`
  //
`

export const Main = styled.header`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.sizes.max};
    display: flex;
    justify-content: center;
    align-items: center;
    color: ${theme.colors.darkText};
    margin: 0 auto;
    font-size: ${theme.fonts.sizes.small};

    /* img {
      margin: 0 auto;
    }
    .header-text-container {
      padding-left: 4rem;
    }
    h1 {
      margin: 7px 0;
    } */

    ${Title} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }

    @media ${theme.media.medium} {
      img {
        display: block;
      }
      p {
        font-size: ${theme.fonts.sizes.small};
      }

      h1 {
        font-size: ${theme.fonts.sizes.medium};
      }

      .header-text-container {
        padding-left: 4px;
      }
    }
  `}
`
export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`
