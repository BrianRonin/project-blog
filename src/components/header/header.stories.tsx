import { Meta, Story } from '@storybook/react/types-6-0'
import { Header, headerProps } from './header'
import { mock_header } from './M.header'

export default {
  title: 'Header',
  component: Header,
  args: mock_header,
  argTypes: {
    //
  },
} as Meta

export const Template: Story<headerProps> = (args) => <Header {...args} />

Template.parameters = {
  backgrounds: {
    default: 'light',
  },
}
