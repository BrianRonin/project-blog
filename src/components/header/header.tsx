import * as S from './S.header'
import { Heading } from '../Heading/index'
import { LogoLink } from '../MENU_NAV_SIDEBAR/logo_link/logo_link'
import { Text } from '../Text'

export type headerProps = {
  alt: string
  srcImg?: string
  allowText?: boolean
  title?: string
  description?: string
}

export const Header = ({
  title,
  description,
  alt,
  srcImg,
  allowText = true,
}: headerProps) => {
  return (
    <S.Main>
      <LogoLink
        text={alt}
        srcImg={srcImg}
        link='/'
      />
      {!!allowText && (
        <S.TextContainer>
          <Heading>{title}</Heading>
          <Text>{description}</Text>
        </S.TextContainer>
      )}
    </S.Main>
  )
}
