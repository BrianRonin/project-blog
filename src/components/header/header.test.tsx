import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/render-theme'
import { Header } from './header'
import { mock_header } from './M.header'

describe('<Header />', () => {
  it('should render with text', () => {
    const { container, debug } = renderTheme(<Header {...mock_header} />)
    expect(container).toBeInTheDocument()
    screen.getByRole('heading', { name: mock_header.title })
    screen.getByText('description lorem ipsum banana couve maÃ§a')
    expect(screen.getByAltText(mock_header.alt)).toHaveAttribute(
      'src',
      mock_header.srcImg,
    )
  })
  it('should render without text', () => {
    const { container, debug } = renderTheme(
      <Header allowText={false} {...mock_header} />,
    )
    expect(container).toBeInTheDocument()
    expect(screen.queryByRole('heading', { name: mock_header.title })).toBe(
      null,
    )
    expect(screen.queryByText(mock_header.description)).toBe(null)
    screen.getByAltText('alt da imagem')
  })
  it('should match snapshot', () => {
    const { container, debug } = renderTheme(<Header {...mock_header} />)
    expect(container).toMatchSnapshot()
  })
})
