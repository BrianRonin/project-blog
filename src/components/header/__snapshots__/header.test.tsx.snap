// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
.c2 {
  color: #000000;
  text-transform: uppercase;
  font-size: 2.4rem;
}

.c6 {
  color: #000000;
  text-transform: none;
  font-size: 6.4rem;
}

.c0 {
  width: 100%;
  max-width: 96rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #333333;
  margin: 0 auto;
  font-size: 1.6rem;
}

.c0 .c1 {
  margin: 0 0 calc(1.6rem - 1rem);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 > img {
  width: 7rem;
  border-radius: 50%;
}

.c4 {
  max-height: 6rem;
}

.c7 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c6 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c0 img {
    display: block;
  }

  .c0 p {
    font-size: 1.6rem;
  }

  .c0 h1 {
    font-size: 2.4rem;
  }

  .c0 .header-text-container {
    padding-left: 4px;
  }
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      <a
        class="c3"
        href="/"
      >
        <img
          alt="alt da imagem"
          class="c4"
          src="/assets/images/piramide.png"
        />
      </a>
    </h1>
    <div
      class="c5"
    >
      <h1
        class="c1 c6"
      >
        um titulo
      </h1>
      <div
        class="c7"
      >
        <p>
          description lorem ipsum banana couve ma√ßa
        </p>
        <p />
      </div>
    </div>
  </header>
</div>
`;
